### ISSUE TITLE
The calculation process of key_padding_mask does not match the document description

### ISSUE THREAD

**Issue by sealoongleft** [pytorch/pytorch#137887]
### code：

> import torch
> import torch.nn as nn
> 
> mha=nn.MultiheadAttention(embed_dim=2, num_heads=1, dropout=0, bias=False, 
>                           add_bias_kv=False,add_zero_attn=False, kdim=None,
>                           vdim=None, batch_first=True,device=None)
> mha.in_proj_weight.data=torch.tensor([[0,1.],
>     [1,1],[0,1],[1,0],[1,1],[0,0]],
>     dtype=torch.float32)
> mha.out_proj.weight.data=torch.tensor([[1,0.],
>     [1,0],[1,1],[0,1]],dtype=torch.float32)
> inputdata=torch.tensor([[1.,0],[0,2]])
> out=mha(inputdata,inputdata,inputdata,average_attn_weights=False,
>         need_weights=True,is_causal=None,attn_mask=None,
>         key_padding_mask=torch.tensor([1.,0],dtype=torch.float)) #bug is here
> print('q|k|v:',inputdata.matmul(mha.in_proj_weight.T))
> print('out:',out)

### result：

> q|k|v: tensor([[0., 1., 0., 1., 1., 0.],
>         [2., 2., 2., 0., 2., 0.]], grad_fn=<MmBackward0>)
> out: (tensor([[1.1535, 1.1535, 1.1535, 0.0000],
>         [1.6021, 1.6021, 1.6021, 0.0000]], grad_fn=<SqueezeBackward1>), tensor([[[0.8465, 0.1535],
>          [0.3979, 0.6021]]], grad_fn=<SqueezeBackward1>))

### description：

> tensor([[[0.8465, 0.1535],
>          [0.3979, 0.6021]]],This result does not match the calculation process described in the API documentation，even if the sum of each line is 1(https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention). The calculation process described in the API documentation is shown in the photo below.
> ![a7a8673caecdad893431a67cb45af784](https://github.com/user-attachments/assets/e1b47944-c4e1-433f-bb37-dbe58b36684f)
### Description of key_padding_mask in API documentation
![image](https://github.com/user-attachments/assets/b6776cf5-3a95-45b0-92bd-11f0b0318a80)
Please take a look at my small bug, technical experts. Thank you very much.

### Versions

win11+intel cpu
cuda12.4
python3.11.9
```[tasklist]
### Tasks
```

cc @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki
```[tasklist]
### Tasks
```

**Comment by sealoongleft** [comment]
anyone can solve this question

**Comment by sealoongleft** [comment]
thanks
